src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -23,101 +23,196 @@ public DataProcessor(List<RoadAccident> roadAccidentList){
+                    minLatitude <= current.getLatitude() && current.getLatitude() <= maxLatitude) {
+                result.add(current);
+            }
+        }
+
+        return result;
     }
 
     /**
      * count incidents by road surface conditions
      * ex:
      * wet -> 2
      * dry -> 5
+     *
      * @return
      */
-    public Map<String, Long> getCountByRoadSurfaceCondition7(){
-        return null;
+    public Map<String, Long> getCountByRoadSurfaceCondition7() {
+        HashMap<String, Long> result = new HashMap<>();
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            String roadSurfaceConditions = current.getRoadSurfaceConditions();
+
+            Long newValue = result.get(roadSurfaceConditions) == null ? 1 : Long.valueOf(result.get(roadSurfaceConditions) + 1);
+            result.put(roadSurfaceConditions, newValue);
+        }
+
+        return result;
     }
 
     /**
      * find the weather conditions which caused the top 3 number of incidents
      * as example if there were 10 accidence in rain, 5 in snow, 6 in sunny and 1 in foggy, then your result list should contain {rain, sunny, snow} - top three in decreasing order
+     *
      * @return
      */
-    public List<String> getTopThreeWeatherCondition7(){
-        return null;
+    public List<String> getTopThreeWeatherCondition7() {
+        List<String> result = new ArrayList<>();
+
+        Map<String, Long> accidentsByWeatherConditions = new HashMap<>();
+
+        Iterator<RoadAccident> roadAccidentIterator = roadAccidentList.iterator();
+
+        while (roadAccidentIterator.hasNext()) {
+            RoadAccident current = roadAccidentIterator.next();
+            String weatherConditions = current.getWeatherConditions();
+
+            Long newValue = accidentsByWeatherConditions.get(weatherConditions) == null ? 1 : Long.valueOf(accidentsByWeatherConditions.get(weatherConditions) + 1);
+            accidentsByWeatherConditions.put(weatherConditions, newValue);
+        }
+
+        List<Map.Entry<String, Long>> listAccidentsByConditions = new LinkedList<Map.Entry<String, Long>>(accidentsByWeatherConditions.entrySet());
+
+        Collections.sort(listAccidentsByConditions, new Comparator<Map.Entry<String, Long>>(){
+
+            @Override
+            public int compare(Map.Entry<String, Long> o1, Map.Entry<String, Long> o2) {
+                // Decreasing order
+                return o2.getValue().compareTo(o1.getValue());
+            }
+        });
+
+        List<Map.Entry<String, Long>> topAccidents = listAccidentsByConditions.subList(0, 3);
+
+        for (Map.Entry<String, Long>  accident: topAccidents) {
+            result.add(accident.getKey());
+        }
+
+        return result;
```

Comment:
Again, your solution is totally fine, but look at [this ](https://github.com/bagedan/EPAM-JMP/pull/3/files#diff-9566c4a7c6afefe7dea1a80cbc457ad8R84)one as well - quite interesting usage of multimap